tests/checkpoint/test_checkpoint_result_format.py

```diff
@@ -193,12 +193,27 @@ def test_sql_result_format_in_checkpoint_pk_defined_one_expectation_complete_out
         checkpoint_name="my_checkpoint",
     )
     evrs: List[ExpectationSuiteValidationResult] = result.list_validation_results()
-    first_result_full_list = evrs[0]["results"][0]["result"]["unexpected_index_list"]
-    assert first_result_full_list == [{"pk_1": 3}, {"pk_1": 4}, {"pk_1": 5}]
-    first_result_partial_list = evrs[0]["results"][0]["result"][
+    index_column_names: List[str] = evrs[0]["results"][0]["result"][
+        "unexpected_index_column_names"
+    ]
+    assert index_column_names == ["pk_1"]
+
+    first_result_full_list: List[Dict[str, Any]] = evrs[0]["results"][0]["result"][
+        "unexpected_index_list"
+    ]
+    assert first_result_full_list == [
+        {"pk_1": 3, "animals": "giraffe"},
+        {"pk_1": 4, "animals": "lion"},
+        {"pk_1": 5, "animals": "zebra"},
+    ]
+    first_result_partial_list: List[Dict[str, Any]] = evrs[0]["results"][0]["result"][
         "partial_unexpected_index_list"
     ]
-    assert first_result_partial_list == [{"pk_1": 3}, {"pk_1": 4}, {"pk_1": 5}]
+    assert first_result_partial_list == [
+        {"pk_1": 3, "animals": "giraffe"},
+        {"pk_1": 4, "animals": "lion"},
+        {"pk_1": 5, "animals": "zebra"},
+    ]
```

Comment:
Not a blocking suggestion, but you might be able to refactor out a lot of this repeated assertion logic into a private module level helper method (like you did with `_add_expectations_and_checkpoint()`).

Reply:
Might open the door to parametrizing all these tests. But your call - only if you think that helps readability.

Reply:
Great suggestion :) If you're ok with it, I'll move the expected values into a separate fixture in this case.